instrumentation/http-url-connection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlHttpAttributesExtractor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.httpurlconnection;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.net.HttpURLConnection;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+class HttpUrlHttpAttributesExtractor
+    extends HttpAttributesExtractor<HttpURLConnection, HttpUrlResponse> {
+  @Override
+  protected String method(HttpURLConnection connection) {
+    return connection.getRequestMethod();
+  }
+
+  @Override
+  protected String url(HttpURLConnection connection) {
+    return connection.getURL().toExternalForm();
+  }
+
+  @Override
+  protected @Nullable String target(HttpURLConnection connection) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable String host(HttpURLConnection connection) {
+    return null;
```

Comment:
Nit: could be
```suggestion
    return connection.getRequestProperty("Host");
```

Reply:
This relates to a general question I've had about instrumeter refactoring: often times the `Instrumenter` is replacing `HttpClientTracer`. `HttpClientTracer` collects a small subset of the attributes available in `HttpAttributesExtractor`. So far, I've been trying to replace `HttpClientTracer` with an implementation of `HttpAttributesExtractor` that collects the same set of attributes, even when it would be straight forward to capture additional ones. I figure better to keep this a strict refactoring, and enhance the behavior by collecting additional attributes in subsequent commits. 

If you agree with keeping this a strict refactoring, then we should collect the host attribute in a followup PR, since its not currently set by `HttpClientTracer`.

Reply:
Sure, that's fine by me 👍